### ISSUE TITLE
near nft contract standards not complete

### ISSUE THREAD

**Issue by evergreen-trading-systems** [near/near-sdk-rs#667]
https://docs.rs/near-contract-standards/4.0.0-pre.4/near_contract_standards/non_fungible_token/index.html

The page says it implements standards described on NEPs https://nomicon.io/Standards/NonFungibleToken/README.html

That's not true, it is missing `RoyaltyPayout`. 

Also missing `Events`. https://github.com/near/near-indexer-for-explorer/blob/f459ca7586aba91cdc31980bdbc2914bcd9391ec/src/db_adapters/assets/nft_types.rs#L13-L14 

It would be really good to fix this so other devs can build upon interoperable standards and not the 80% compliant 20% siloed it is today.

**Comment by evergreen-trading-systems** [comment]
https://github.com/near/near-sdk-rs/issues/649 / https://github.com/near/near-sdk-rs/pull/627- addresses events

however royalty payout is missing

**Comment by austinabell** [comment]
Yes, this is the case that the standards impl lag behind the spec, and that is just because some things need time to be stabilized outside of this repo or just time to be implemented carefully because breaking changes can be very inconvenient for developers. At the end of the day, these standards impl here are just a framework for building these things quicker, not the source of truth.

As you mentioned there are PRs for the events which are being iterated on, and royalty payout will come soon but no certain timeline yet as it's a new one and I'm not sure where in the ecosystem it is being used.

**Comment by willemneal** [comment]
@evergreen-trading-systems  See #673 

I also think the standards process has suffered because there wasn't an accompanying implementation.  Even if it's not the source of truth, it is important in catching bugs especially since we've chosen rust types as the current standards, though I'd like to see it move to `.wit`.

For example, in the payouts standard, the memo field is missing `nft_transfer_payout` even though it's needed by `nft_transfer`.

**Comment by r-near** [comment]
Resolved — NEP-171 events are implemented at `near-contract-standards/src/non_fungible_token/events.rs` (NftMint/NftTransfer/NftBurn). Royalties/payout (NEP-199) is intentionally separate; please open a fresh focused issue if it's desired.
